flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/reader/AvroGenericRecordReaderFunction.java

```diff
@@ -72,7 +73,8 @@ public AvroGenericRecordReaderFunction(
     this.io = io;
     this.encryption = encryption;
     this.rowDataReader =
-        new RowDataFileScanTaskReader(tableSchema, readSchema, nameMapping, caseSensitive);
+        new RowDataFileScanTaskReader(
+            tableSchema, readSchema, nameMapping, caseSensitive, Collections.emptyList());
```

Comment:
this constructor needs to take the filters and pass the arg here.

Reply:
Done!